flow/connectors/postgres/cdc.go

```diff
@@ -353,6 +353,21 @@ func (p *PostgresCDCSource) consumeStream(
 						tablePKeyLastSeen[tablePkeyVal] = len(localRecords) - 1
 					}
 				case *model.DeleteRecord:
+					compositePKeyString, err := p.compositePKeyToString(req, rec)
+					if err != nil {
+						return err
+					}
+
+					tablePkeyVal := model.TableWithPkey{
+						TableName:  tableName,
+						PkeyColVal: compositePKeyString,
+					}
+					_, ok := tablePKeyLastSeen[tablePkeyVal]
+					if ok {
+						latestRecord := localRecords[tablePKeyLastSeen[tablePkeyVal]]
```

Comment:
`tablePKeyLastSeen[tablePkeyVal]` could've been stored to a local variable on line 365